lib/puppet/configurer.rb

```diff
@@ -44,6 +44,12 @@ def init_storage
     end
   end
 
+  def init_http_pool_monitoring(report, http_pool)
+    if http_pool.is_a? Puppet::Network::HTTP::Pool
+      http_pool.report = report
```

Comment:
I don't like storing the report in the pool this way. Once the run finishes, the pool still has a reference to the now obsolete report.